includes/abstracts/abstract.llms.post.model.php

```diff
@@ -910,70 +911,130 @@ protected function scrub_field( $val, $type ) {
 	 *
 	 * @since 3.0.0
 	 * @since 3.30.3 Use `wp_slash()` when setting properties.
+	 * @since [version] Turned to be only a wrapper for the update_model() method.
 	 *
-	 * @param string $key Key of the property.
-	 * @param mixed  $val Value to set the property with.
+	 * @param string|array $key_or_array Key of the property or a an associative array of key/val pairs.
+	 * @param mixed        $val          Optional. Value to set the property with. Default empty string.
+	 *                                   This parameter will be ignored when the first parameter is an associative array of key/val pairs.
 	 * @return boolean true on success, false on error or if the submitted value is the same as what's in the database
 	 */
-	public function set( $key, $val ) {
+	public function set( $key_or_array, $val = '' ) {
 
-		$val = $this->scrub( $key, $val );
+		$model_array = array();
+		if ( ! is_array( $key_or_array ) ) {
+			$model_array = array(
+				$key_or_array => $val,
+			);
+		} else {
+			$model_array = $key_or_array;
+		}
+		return $this->update_model( $model_array );
 
-		// update WordPress Post Properties using the wp_insert_post() function
-		if ( in_array( $key, array_keys( $this->get_post_properties() ) ) ) {
+	}
 
-			$post_key = 'post_' . $key;
 
-			switch ( $key ) {
+	/**
+	 * Updater
+	 *
+	 * @since [version]
+	 *
+	 * @param array $model_array Associative array of key/val pairs.
+	 * @return boolean true on success, false on error or if there was nothing to update.
+	 */
+	public function update_model( $model_array ) {
```

Comment:
I think the method should be called `set_bulk()`

I also think that we should add an additional parameter `$wp_error` so we can grab **more information** in an error as an option.

This parameter should default to `false` and when `false` it will just return a simple boolean when an error / no update is required (exactly how the existing `set()` method works. Also allowing for backwards compat.

If `$wp_error' is `true` then we can create a `WP_Error` object and collect as many errors are encountered and then return that error object (instead of false).

The REST API will likely wish to use this to provide more robust information during an create/update request but existing methods which are likely just checking for a bool on set can safely continue working as is.

Do you think that solves the error problem you're asking about?

Reply:
Yes totally, thanks!

Reply:
in 565950b